src/EditorFeatures/CSharpTest/AssignOutParameters/AssignOutParametersAtStartTests.cs

```diff
@@ -0,0 +1,632 @@
+    {
+        internal override (DiagnosticAnalyzer, CodeFixProvider) CreateDiagnosticProviderAndFixer(Workspace workspace)
+            => (null, new AssignOutParametersAtStartCodeFixProvider());
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestForSimpleReturn()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    char M(out int i)
+    {
+        [|return 'a';|]
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestForSwitchSectionReturn()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    char M(out int i)
+    {
+        switch (0)
+        {
+            default:
+                [|return 'a';|]
+        }
+    }
+}",
+@"class C
+{
+    char M(out int i)
+    {
+        i = 0;
+        switch (0)
+        {
+            default:
+                return 'a';
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestMissingWhenVariableAssigned()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    char M(out int i)
+    {
+        i = 0;
+        [|return 'a';|]
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestWhenNotAssignedThroughAllPaths1()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    char M(bool b, out int i)
+    {
+        if (b)
+            i = 1;
+
+        [|return 'a';|]
+    }
+}",
+@"class C
+{
+    char M(bool b, out int i)
+    {
+        i = 0;
+        if (b)
+            i = 1;
+
+        return 'a';
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestWhenNotAssignedThroughAllPaths2()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    bool M(out int i1, out int i2)
+    {
+        [|return Try(out i1) || Try(out i2);|]
+    }
+
+    bool Try(out int i)
+    {
+        i = 0;
+        return true;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestMissingWhenAssignedThroughAllPaths()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    char M(out int i)
+    {
+        if (b)
+            i = 1;
+        else
+            i = 2;
+        
+        [|return 'a';|]
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestMultiple()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    char M(out int i, out string s)
+    {
+        [|return 'a';|]
```

Comment:
>return [](start = 10, length = 6)

shouldn't we expect assignments to `i` and `s`?

Reply:
same reason as above :)